interface/rest/shops_handler.go

```diff
@@ -16,18 +17,29 @@ func NewShopsHandler(searchShopService search_shop.Service) *shopsHandler {
 }
 
 func (s *shopsHandler) Search(w http.ResponseWriter, req *http.Request) {
-	// validation
-	// logging
+	// TODO: logging
 	query := req.URL.Query()
+
+	// TODO: validation all inputs
 	search := query.Get("query")
 	lat, _ := strconv.ParseFloat(query.Get("lat"), 32)
 	long, _ := strconv.ParseFloat(query.Get("long"), 32)
 
+	if len(search) == 0 {
+		// TODO: make error object
+		respond(w, http.StatusBadRequest, "must provide a search query")
```

Comment:
Like you mentioned here in your todo, it'd be good to have an error object that has a code, then you can use that code to perform translations on the frontend.

Reply:
I also left some other feedback about the searching bit in your REST definition.